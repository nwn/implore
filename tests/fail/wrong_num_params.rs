use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op]
fn neg(x: &mut X, y: u32) {}

#[impl_op]
fn mul(x: &X) {}

#[impl_op]
fn add_assign(x: &mut X, y: u32, z: bool) {}

#[impl_op]
fn index_mut(x: &mut X, y: u32, z: bool) -> &mut X {}

#[impl_op]
fn deref() {}

fn main() {}
