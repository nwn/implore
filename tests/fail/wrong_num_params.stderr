error: operation `Neg` takes exactly 1 argument, found 2
 --> $DIR/wrong_num_params.rs:7:7
  |
7 | fn neg(x: &mut X, y: u32) {}
  |       ^^^^^^^^^^^^^^^^^^^

error: operation `Mul` takes exactly 2 arguments, found 1
  --> $DIR/wrong_num_params.rs:10:7
   |
10 | fn mul(x: &X) {}
   |       ^^^^^^^

error: operation `AddAssign` takes exactly 2 arguments, found 3
  --> $DIR/wrong_num_params.rs:13:14
   |
13 | fn add_assign(x: &mut X, y: u32, z: bool) {}
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: operation `IndexMut` takes exactly 2 arguments, found 3
  --> $DIR/wrong_num_params.rs:16:13
   |
16 | fn index_mut(x: &mut X, y: u32, z: bool) -> &mut X {}
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: operation `Deref` takes exactly 1 argument, found 0
  --> $DIR/wrong_num_params.rs:19:9
   |
19 | fn deref() {}
   |         ^^
